{"primaryContentSections":[{"kind":"declarations","declarations":[{"tokens":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"kind":"identifier","text":"cancel"},{"kind":"text","text":"("},{"kind":"externalParam","text":"promise"},{"kind":"text","text":": "},{"kind":"typeIdentifier","identifier":"doc:\/\/org.swift.docc\/documentation\/NIO\/EventLoopPromise","preciseIdentifier":"s:3NIO16EventLoopPromiseV","text":"EventLoopPromise"},{"kind":"text","text":"<"},{"kind":"typeIdentifier","text":"Void","preciseIdentifier":"s:s4Voida"},{"kind":"text","text":">? = nil)"}],"languages":["swift"],"platforms":["macOS"]}]},{"kind":"content","content":[{"anchor":"discussion","level":2,"type":"heading","text":"Discussion"},{"type":"paragraph","inlineContent":[{"type":"text","text":"Whether the execution of the task is immediately canceled depends on whether the execution of a task has already begun."},{"type":"text","text":" "},{"type":"text","text":"This means immediate cancellation is not guaranteed."}]},{"type":"paragraph","inlineContent":[{"type":"text","text":"The safest way to cancel is by using the passed reference of "},{"type":"codeVoice","code":"RepeatedTask"},{"type":"text","text":" inside the task closure."}]},{"type":"paragraph","inlineContent":[{"type":"text","text":"If the promise parameter is not "},{"type":"codeVoice","code":"nil"},{"type":"text","text":", the passed promise is fulfilled when cancellation is complete."},{"type":"text","text":" "},{"type":"text","text":"Passing a promise does not prevent fulfillment of any promise provided on original task creation."}]}]}],"schemaVersion":{"major":0,"minor":3,"patch":0},"sections":[],"variants":[{"paths":["\/documentation\/nio\/repeatedtask\/cancel(promise:)"],"traits":[{"interfaceLanguage":"swift"}]}],"identifier":{"url":"doc:\/\/org.swift.docc\/documentation\/NIO\/RepeatedTask\/cancel(promise:)","interfaceLanguage":"swift"},"abstract":[{"type":"text","text":"Try to cancel the execution of the repeated task."}],"kind":"symbol","metadata":{"fragments":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"kind":"identifier","text":"cancel"},{"kind":"text","text":"("},{"kind":"externalParam","text":"promise"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"EventLoopPromise","preciseIdentifier":"s:3NIO16EventLoopPromiseV"},{"kind":"text","text":"<"},{"kind":"typeIdentifier","text":"Void","preciseIdentifier":"s:s4Voida"},{"kind":"text","text":">?)"}],"title":"cancel(promise:)","roleHeading":"Instance Method","role":"symbol","symbolKind":"method","externalID":"s:3NIO12RepeatedTaskC6cancel7promiseyAA16EventLoopPromiseVyytGSg_tF","modules":[{"name":"NIO"}]},"hierarchy":{"paths":[["doc:\/\/org.swift.docc\/documentation\/NIO","doc:\/\/org.swift.docc\/documentation\/NIO\/RepeatedTask"]]},"references":{"doc://org.swift.docc/documentation/NIO":{"role":"collection","title":"NIO","abstract":[],"identifier":"doc:\/\/org.swift.docc\/documentation\/NIO","kind":"symbol","type":"topic","url":"\/documentation\/nio"},"doc://org.swift.docc/documentation/NIO/EventLoopPromise":{"role":"symbol","title":"EventLoopPromise","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"EventLoopPromise"}],"abstract":[{"type":"text","text":"A promise to provide a result later."}],"identifier":"doc:\/\/org.swift.docc\/documentation\/NIO\/EventLoopPromise","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"EventLoopPromise"}],"url":"\/documentation\/nio\/eventlooppromise"},"doc://org.swift.docc/documentation/NIO/RepeatedTask/cancel(promise:)":{"role":"symbol","title":"cancel(promise:)","fragments":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"kind":"identifier","text":"cancel"},{"kind":"text","text":"("},{"kind":"externalParam","text":"promise"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"EventLoopPromise","preciseIdentifier":"s:3NIO16EventLoopPromiseV"},{"kind":"text","text":"<"},{"kind":"typeIdentifier","text":"Void","preciseIdentifier":"s:s4Voida"},{"kind":"text","text":">?)"}],"abstract":[{"type":"text","text":"Try to cancel the execution of the repeated task."}],"identifier":"doc:\/\/org.swift.docc\/documentation\/NIO\/RepeatedTask\/cancel(promise:)","kind":"symbol","type":"topic","url":"\/documentation\/nio\/repeatedtask\/cancel(promise:)"},"doc://org.swift.docc/documentation/NIO/RepeatedTask":{"role":"symbol","title":"RepeatedTask","fragments":[{"kind":"keyword","text":"class"},{"kind":"text","text":" "},{"kind":"identifier","text":"RepeatedTask"}],"abstract":[{"type":"text","text":"Returned once a task was scheduled to be repeatedly executed on the "},{"type":"codeVoice","code":"EventLoop"},{"type":"text","text":"."}],"identifier":"doc:\/\/org.swift.docc\/documentation\/NIO\/RepeatedTask","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"RepeatedTask"}],"url":"\/documentation\/nio\/repeatedtask"}}}